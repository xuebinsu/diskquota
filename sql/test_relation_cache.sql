begin;
create table t(i int);
insert into t select generate_series(1, 100000);

WITH table_size AS (
    SELECT diskquota.diskquota_fetch_table_stat(1, ARRAY['t'::regclass]) AS a
    FROM  gp_dist_random('gp_id')
)
SELECT (a).* FROM table_size order by table_size;

select pg_table_size('t') as ts from gp_dist_random('gp_id') order by ts;
abort;

begin;
create table t(t text);
insert into t select array(select * from generate_series(1,1000)) from generate_series(1, 1000);

WITH table_size AS (
    SELECT diskquota.diskquota_fetch_table_stat(1, ARRAY['t'::regclass]) AS a
    FROM  gp_dist_random('gp_id')
)
SELECT (a).* FROM table_size order by table_size;

select pg_table_size('t') as ts from gp_dist_random('gp_id') order by ts;
abort;
