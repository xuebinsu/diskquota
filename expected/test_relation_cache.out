begin;
create table t(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into t select generate_series(1, 100000);
WITH table_size AS (
    SELECT diskquota.diskquota_fetch_table_stat(1, ARRAY['t'::regclass]) AS a
    FROM  gp_dist_random('gp_id')
)
SELECT (a).* FROM table_size order by table_size;
 TABLE_OID | TABLE_SIZE | GP_SEGMENT_ID 
-----------+------------+---------------
     16458 |    1310720 |             0
     16458 |    1310720 |             1
     16458 |    1310720 |             2
(3 rows)

select pg_table_size('t') as ts from gp_dist_random('gp_id') order by ts;
   ts    
---------
 1310720
 1310720
 1310720
(3 rows)

abort;
begin;
create table t(t text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into t select array(select * from generate_series(1,1000)) from generate_series(1, 1000);
WITH table_size AS (
    SELECT diskquota.diskquota_fetch_table_stat(1, ARRAY['t'::regclass]) AS a
    FROM  gp_dist_random('gp_id')
)
SELECT (a).* FROM table_size order by table_size;
 TABLE_OID | TABLE_SIZE | GP_SEGMENT_ID 
-----------+------------+---------------
     16461 |      32768 |             0
     16461 |      32768 |             2
     16461 |    4227072 |             1
(3 rows)

select pg_table_size('t') as ts from gp_dist_random('gp_id') order by ts;
   ts    
---------
   32768
   32768
 4227072
(3 rows)

abort;
